src/Umbraco.Core/Services/ContentService.cs

```diff
@@ -2008,6 +2008,8 @@ private Attempt<PublishStatus> SaveAndPublishDo(IContent content, int userId = 0
                     content.WriterId = userId;
 
                     repository.AddOrUpdate(content);
+                    
+                    content = repository.Get(content.Id);

```

Comment:
The thing about changing that AddOrUpdate method is that you would have to change the interfaces for it - affecting all repositories. Since thats a rather significant change, I figured just doing a get after a save wouldn't be the worst way to get around that for now.